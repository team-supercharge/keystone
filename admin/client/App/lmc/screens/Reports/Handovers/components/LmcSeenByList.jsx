import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import _ from 'lodash'

export class LmcSeenByList extends Component {
    state = {
        isExpanded: false
    }

    toggleExpand = () => {
        this.setState(prevState => ({ 
            isExpanded: !prevState.isExpanded 
        }));
    }

    renderPicture = (carer) => {
        return (
            <div 
                key={carer.id} 
                style={styles.rows}
            >
                <div 
                    className='lmc-profile-picture__handover__small' 
                    style={{
                        ...styles.image,
                        background: `url(${carer.picture || PLACEHOLDER_IMAGE})`
                    }}
                />
                <span 
                    style={this.state.isExpanded 
                        ? styles.textColumn 
                        : styles.textRow}
                >
                    {carer.name.first} {carer.name.last}
                </span>
            </div>
        )
    }

    render () {
        const { seenBy } = this.props;
        const { isExpanded } = this.state;
        
        if (!seenBy || !seenBy.length) return null

        return (
            <div style={styles.mainContainer}>
                <div 
                    style={{
                        ...styles.containerSeenBy, 
                        flexDirection: 'row' 
                    }} 
                    onClick={() => this.toggleExpand()}
                >
                    <span style={styles.seen}>
                        Seen by
                    </span>
                    <div
                        style={{
                            display: 'flex',
                            flexDirection: isExpanded ? 'column' : 'row',
                            flexWrap: 'wrap',
                        }}
                    >
                        { seenBy.map(carer => {
                            return (
                                <div 
                                    key={carer.id}
                                    style={isExpanded 
                                        ? styles.imageContainerColumn 
                                        : null }
                                >
                                    { this.renderPicture(carer) }
                                </div>
                            )
                        }) }
                    </div>
                </div>
            </div>
        );  
    }
}

const styles = {
    containerSeenBy: {
        display: 'flex',
        cursor: 'pointer',
        padding: '20px 0px 20px 0px'
    },
    seen: {
        paddingTop: 8,
        opacity: 0.8,
        fontSize: 12,
        marginRight: 10,
        flexShrink: 0,
    },
    imageContainerColumn: {
        padding: '0px 4px 2px 0px'
    },
    image: {
        marginRight: 2,
    },
    rows: {
        display: 'flex',
        flexDirection: 'row',
    },
    textRow: {
        display: 'none',
    },
    textColumn: {
        fontSize: 12,
        opacity: 0.6,
        padding: '8px 0px 0px 10px'
    }
}

const PLACEHOLDER_IMAGE = 'https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png';

LmcSeenByList.propTypes = {
    carers: PropTypes.array,
    seenBy: PropTypes.array.isRequired
}

const mapStateToProps = (state) => {
    return {
        carers: state.data.carers
    }
}   

export default connect(mapStateToProps, null)(LmcSeenByList)