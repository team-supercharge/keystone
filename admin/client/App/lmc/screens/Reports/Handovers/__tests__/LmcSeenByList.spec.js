import React from 'react'
import { shallow } from 'enzyme'
import { LmcSeenByList } from '../components/LmcSeenByList.jsx'

describe('LmcSeenByList', () => {
    let wrapper
    let carers

    beforeEach(() => {
        carers = [
            { id: 'id1', name: { first: 'test', last: 'carer' }},
            { id: 'id2', name: { first: 'test', last: 'carer2' }},
            { id: 'id3', name: { first: 'test', last: 'carer3' }}
        ]
        wrapper = shallow(
            <LmcSeenByList 
                carers={carers}
                seenBy={carers}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
})