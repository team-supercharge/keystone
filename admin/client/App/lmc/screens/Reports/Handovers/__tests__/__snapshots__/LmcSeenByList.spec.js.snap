// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcSeenByList renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
          "display": "flex",
          "flexDirection": "row",
          "padding": "20px 0px 20px 0px",
        }
      }
    >
      <span
        style={
          Object {
            "flexShrink": 0,
            "fontSize": 12,
            "marginRight": 10,
            "opacity": 0.8,
            "paddingTop": 8,
          }
        }
      >
        Seen by
      </span>
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "row",
            "flexWrap": "wrap",
          }
        }
      >
        <div
          style={null}
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover__small"
              style={
                Object {
                  "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                  "marginRight": 2,
                }
              }
            />
            <span
              style={
                Object {
                  "display": "none",
                }
              }
            >
              test
               
              carer
            </span>
          </div>
        </div>
        <div
          style={null}
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover__small"
              style={
                Object {
                  "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                  "marginRight": 2,
                }
              }
            />
            <span
              style={
                Object {
                  "display": "none",
                }
              }
            >
              test
               
              carer2
            </span>
          </div>
        </div>
        <div
          style={null}
        >
          <div
            style={
              Object {
                "display": "flex",
                "flexDirection": "row",
              }
            }
          >
            <div
              className="lmc-profile-picture__handover__small"
              style={
                Object {
                  "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                  "marginRight": 2,
                }
              }
            />
            <span
              style={
                Object {
                  "display": "none",
                }
              }
            >
              test
               
              carer3
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <div
        onClick={[Function]}
        style={
          Object {
            "cursor": "pointer",
            "display": "flex",
            "flexDirection": "row",
            "padding": "20px 0px 20px 0px",
          }
        }
      >
        <span
          style={
            Object {
              "flexShrink": 0,
              "fontSize": 12,
              "marginRight": 10,
              "opacity": 0.8,
              "paddingTop": 8,
            }
          }
        >
          Seen by
        </span>
        <div
          style={
            Object {
              "display": "flex",
              "flexDirection": "row",
              "flexWrap": "wrap",
            }
          }
        >
          <div
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <div
                className="lmc-profile-picture__handover__small"
                style={
                  Object {
                    "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                    "marginRight": 2,
                  }
                }
              />
              <span
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                test
                 
                carer
              </span>
            </div>
          </div>
          <div
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <div
                className="lmc-profile-picture__handover__small"
                style={
                  Object {
                    "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                    "marginRight": 2,
                  }
                }
              />
              <span
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                test
                 
                carer2
              </span>
            </div>
          </div>
          <div
            style={null}
          >
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                }
              }
            >
              <div
                className="lmc-profile-picture__handover__small"
                style={
                  Object {
                    "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                    "marginRight": 2,
                  }
                }
              />
              <span
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                test
                 
                carer3
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcSeenByList
        carers={
          Array [
            Object {
              "id": "id1",
              "name": Object {
                "first": "test",
                "last": "carer",
              },
            },
            Object {
              "id": "id2",
              "name": Object {
                "first": "test",
                "last": "carer2",
              },
            },
            Object {
              "id": "id3",
              "name": Object {
                "first": "test",
                "last": "carer3",
              },
            },
          ]
        }
        seenBy={
          Array [
            Object {
              "id": "id1",
              "name": Object {
                "first": "test",
                "last": "carer",
              },
            },
            Object {
              "id": "id2",
              "name": Object {
                "first": "test",
                "last": "carer2",
              },
            },
            Object {
              "id": "id3",
              "name": Object {
                "first": "test",
                "last": "carer3",
              },
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcSeenByList {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "carers": Array [
            Object {
              "id": "id1",
              "name": Object {
                "first": "test",
                "last": "carer",
              },
            },
            Object {
              "id": "id2",
              "name": Object {
                "first": "test",
                "last": "carer2",
              },
            },
            Object {
              "id": "id3",
              "name": Object {
                "first": "test",
                "last": "carer3",
              },
            },
          ],
          "seenBy": Array [
            Object {
              "id": "id1",
              "name": Object {
                "first": "test",
                "last": "carer",
              },
            },
            Object {
              "id": "id2",
              "name": Object {
                "first": "test",
                "last": "carer2",
              },
            },
            Object {
              "id": "id3",
              "name": Object {
                "first": "test",
                "last": "carer3",
              },
            },
          ],
        },
        "refs": Object {},
        "renderPicture": [Function],
        "state": Object {
          "isExpanded": false,
        },
        "toggleExpand": [Function],
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "20px 0px 20px 0px",
              }
            }
          >
            <span
              style={
                Object {
                  "flexShrink": 0,
                  "fontSize": 12,
                  "marginRight": 10,
                  "opacity": 0.8,
                  "paddingTop": 8,
                }
              }
            >
              Seen by
            </span>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer
                  </span>
                </div>
              </div>
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer2
                  </span>
                </div>
              </div>
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <div
            onClick={[Function]}
            style={
              Object {
                "cursor": "pointer",
                "display": "flex",
                "flexDirection": "row",
                "padding": "20px 0px 20px 0px",
              }
            }
          >
            <span
              style={
                Object {
                  "flexShrink": 0,
                  "fontSize": 12,
                  "marginRight": 10,
                  "opacity": 0.8,
                  "paddingTop": 8,
                }
              }
            >
              Seen by
            </span>
            <div
              style={
                Object {
                  "display": "flex",
                  "flexDirection": "row",
                  "flexWrap": "wrap",
                }
              }
            >
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer
                  </span>
                </div>
              </div>
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer2
                  </span>
                </div>
              </div>
              <div
                style={null}
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <div
                    className="lmc-profile-picture__handover__small"
                    style={
                      Object {
                        "background": "url(https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/profile_pic_placeholder.png)",
                        "marginRight": 2,
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "display": "none",
                      }
                    }
                  >
                    test
                     
                    carer3
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcSeenByList
    carers={
      Array [
        Object {
          "id": "id1",
          "name": Object {
            "first": "test",
            "last": "carer",
          },
        },
        Object {
          "id": "id2",
          "name": Object {
            "first": "test",
            "last": "carer2",
          },
        },
        Object {
          "id": "id3",
          "name": Object {
            "first": "test",
            "last": "carer3",
          },
        },
      ]
    }
    seenBy={
      Array [
        Object {
          "id": "id1",
          "name": Object {
            "first": "test",
            "last": "carer",
          },
        },
        Object {
          "id": "id2",
          "name": Object {
            "first": "test",
            "last": "carer2",
          },
        },
        Object {
          "id": "id3",
          "name": Object {
            "first": "test",
            "last": "carer3",
          },
        },
      ]
    }
  />,
}
`;
